Quadriga/src/main/java/edu/asu/spring/quadriga/rest/PassThroughProjectRestController.java

```diff
@@ -0,0 +1,103 @@
+import edu.asu.spring.quadriga.service.passthroughproject.IPassThroughProjectManager;
+
+/**
+ * 
+ * Controller for pass through projects rest apis exposed to other clients
+ *
+ */
+@Controller
+public class PassThroughProjectRestController {
+
+    @Autowired
+    private IPassThroughProjectManager passThroughProjectManager;
+
+    @Autowired
+    private IUserManager userManager;
+
+    @Autowired
+    private IRestMessage errorMessageRest;
+
+    @Autowired
+    private IPassThroughProjectDocumentReader passThroughProjectDocumentReader;
+
+    @Autowired
+    private INetworkManager networkManager;
+
+    @RequestMapping(value = "rest/passthroughproject", method = RequestMethod.POST)
+    public ResponseEntity<String> getPassThroughProject(HttpServletRequest request,
+            @RequestHeader("Accept") String accept, HttpServletResponse response, @RequestBody String xml,
+            Principal principal) throws RestException {
+
+        xml = xml.trim();
+        if (xml.isEmpty()) {
+            String errorMsg = errorMessageRest.getErrorMsg("Please provide XML in body of the post request.");
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+
+        }
+        String userid = principal.getName();
+        String networkId = null;
+        try {
+            IUser user = userManager.getUser(userid);
+            PassThroughProjectInfo passThroughProjectInfo = passThroughProjectDocumentReader
+                    .getPassThroughProjectInfo(xml);
+
+            IPassThroughProject project = passThroughProjectDocumentReader
+                    .getPassThroughProject(passThroughProjectInfo);
+
+            String projectId = passThroughProjectManager.savePassThroughProject(user, project);
+
+            String workspaceId = passThroughProjectManager.createWorkspaceForExternalProject(passThroughProjectInfo,
+                    projectId, user);
+
+            String network = passThroughProjectDocumentReader.getNetwork(xml);
+
+            String responseFromQStore = networkManager.storeNetworks(network);
+
+            networkId = networkManager.storeNetworkDetails(responseFromQStore, user,
+                    passThroughProjectInfo.getNetworkName(), workspaceId, INetworkManager.NEWNETWORK, "",
+                    INetworkManager.VERSION_ZERO);
+        } catch (QStoreStorageException | DocumentParserException | QuadrigaStorageException | NoSuchRoleException
+                | JAXBException | QuadrigaAccessException e) {
+            String errorMsg = errorMessageRest.getErrorMsg(e.getMessage());
+            return new ResponseEntity<String>(errorMsg, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.setContentType(MediaType.APPLICATION_XML);
+        httpHeaders.set("networkid", networkId);
+
+        return new ResponseEntity<String>("Success", httpHeaders, HttpStatus.OK);
```

Comment:
the method should return project id, workspace id, project name, workspace name, and if a network was created its id. You can simply return that as json.